proxy/http/HttpTransact.cc

```diff
@@ -5905,6 +5906,12 @@ HttpTransact::is_stale_cache_response_returnable(State *s)
   if (!s->cache_info.directives.does_client_permit_lookup) {
     return false;
   }
+  // We don't serve a stale negative cache.
```

Comment:
This might have a pretty big impact in other places where calls this function.

@sudheerv Is this change safe with `CACHE_WL_FAIL_ACTION_STALE_ON_REVALIDATE`? 
https://github.com/apache/trafficserver/blob/690ec54950611a1955b34911b73a222365998667/proxy/http/HttpTransact.cc#L7215-L7216

Reply:
@fdiary How about move this change under the if-block of doing negative revalidating to minimize side effects? 
https://github.com/apache/trafficserver/blob/b6b61fbd3e232dbef010dc8b712bb2076510bd7f/proxy/http/HttpTransact.cc#L4347

My concern is this changes behavior of setting `2` and `3` of  `proxy.config.http.cache.open_write_fail_action`[*1]. It's a config to do stale-while-revalidate, so if we never serve negative stale cache, all simultaneous requests might go to the origin server.

[*1] https://docs.trafficserver.apache.org/en/latest/admin-guide/files/records.config.en.html#proxy-config-http-cache-open-write-fail-action

Reply:
@masaori335 Thanks for your proposal, that is reasonable. I force-updated the change in this PR.